plugin/trino-mongodb/src/main/java/io/trino/plugin/mongodb/MongoMetadata.java

```diff
@@ -619,19 +627,81 @@ public Optional<ConstraintApplicationResult<ConnectorTableHandle>> applyFilter(C
             remainingFilter = TupleDomain.withColumnDomains(unsupported);
         }
 
-        if (oldDomain.equals(newDomain)) {
+        ConnectorExpression oldExpression = constraint.getExpression();
```

Comment:
Please use ConnectorExpressionRewriter instead. You can refer to PostgreSqlClient class.

Reply:
Implement contains function pushdown using ConnectorExpressionRewriter.